caf-audit/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditConnection.java

```diff
@@ -30,7 +30,29 @@
 
     public ElasticAuditConnection(ConfigurationSource configSource) throws ConfigurationException {
         //  Get Elasticsearch configuration.
-        final ElasticAuditConfiguration config = configSource.getConfiguration(ElasticAuditConfiguration.class);
+        final ElasticAuditConfiguration config;
+        if (configSource == null) {
+            config = new ElasticAuditConfiguration();
+
+            // Get the Elasticsearch host and port values from env var
+            config.setHostAndPortValues(System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES, System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES)));
+
+            // Get the Elasticsearch cluster name from env var else default to "elasticsearch-cluster"
+            String clusterName = System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_CLUSTER_NAME, System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_CLUSTER_NAME));
```

Comment:
I don't think we should be defaulting this within the code.  Smart defaults should be set with the configuration files but not within the code.
The same probably goes for the Shard and Replicas below also.